src/browser/BrowserService.cpp

```diff
@@ -618,9 +622,27 @@ bool BrowserService::updateEntry(const QString& dbid,
                 MessageBox::Save | MessageBox::Cancel,
                 MessageBox::Cancel,
                 MessageBox::Raise);
+        } else if (isReferencePasswordChanged) {
+            raiseWindow();
+            dialogResult =
+                MessageBox::question(nullptr,
+                                     tr("KeePassXC: Update Entry"),
+                                     tr("Do you want to update the information in %1 - %2?\n\nThe selected entry is a "
+                                        "reference. Select Overwrite to change password to the original entry.\nSelect "
+                                        "Save to change the password of the current referenced entry.")
+                                         .arg(QUrl(siteUrlStr).host(), username),
+                                     MessageBox::Save | MessageBox::Cancel | MessageBox::Overwrite,
+                                     MessageBox::Cancel,
+                                     MessageBox::Raise);
```

Comment:
I would think that `overwrite` would replace the reference with the new password (ie, the generally undesired option). `Save` would save the new password to the original entry. We might also want to use `Keep` to keep the reference and save the password to the original entry. Thoughts @phoerious ?

Reply:
Made the changes as you suggested. Makes more sense.